client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/entities/requests/RegisterSchemaRequest.java

```diff
@@ -58,31 +59,18 @@ public boolean equals(Object o) {
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    if (!super.equals(o)) {
-      return false;
-    }
-
     RegisterSchemaRequest that = (RegisterSchemaRequest) o;
-
-    if (schema != null ? !schema.equals(that.schema) : that.schema != null) {
-      return false;
-    }
-
-    return true;
+    return id == that.id && Objects.equals(schema, that.schema);
```

Comment:
🙏 thanks for cleaning up the logic here, this method was pretty hard to read as-was!